src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs

```diff
@@ -82,6 +82,18 @@ public class NewAnalysisServicesServer : AnalysisServicesCmdletBase
             HelpMessage = "Firewall configuration")]
         public PsAzureAnalysisServicesFirewallConfig FirewallConfig { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false,
+            HelpMessage = "Gateway resource name")]
```

Comment:
You should allow them to specify the gateway by resource ID instead, they are much more likely to have this value

Reply:
We construct the resource ID internally, but sure.